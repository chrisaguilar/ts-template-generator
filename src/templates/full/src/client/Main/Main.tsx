import * as React from 'react';
const styles = require('./MainStyle.scss');

interface MainProps {
  message: string;
}

export class Main extends React.Component<MainProps, any> {
  constructor(props: MainProps) {
    super(props);
  }

  render() {
    return (
      <div className={ styles.main }>
        <h1 className={ styles.text }>{ this.props.message }</h1>
      </div>
    );
  }
}
