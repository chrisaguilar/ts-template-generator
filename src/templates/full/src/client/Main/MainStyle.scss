.main {
  background-color: chartreuse;
  padding: 1em;
  min-height: 100vh;
  .text {
    background-color: black;
    color: yellow;
    padding: 2em;
    text-align: center;
  }
}
