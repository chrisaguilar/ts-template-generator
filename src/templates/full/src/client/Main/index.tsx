import * as React from 'react';

const styles = require('./styles.scss');

interface MainProps {
  message: string;
}

export default (props: MainProps) => {
  return (
    <div>
      <h3>{ props.message }</h3>
    </div>
  );
};
