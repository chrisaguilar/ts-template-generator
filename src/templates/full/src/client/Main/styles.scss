.main {
  background: orange;
  font-size: 0.5em;
  font-style: italic;
}
