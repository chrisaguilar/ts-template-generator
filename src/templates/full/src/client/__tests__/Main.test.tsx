import { shallow } from 'enzyme';
import * as React from 'react';
import { create as snapshot } from 'react-test-renderer';

import Main from '../Main';

describe('<Main message={string} />', () => {
  it('should render', () => {
    const wrapper = shallow(<Main message='' />);
    expect(wrapper.length).toBeGreaterThan(0);
  });
  it('renders correctly', () => {
    const tree = snapshot(<Main message='' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
