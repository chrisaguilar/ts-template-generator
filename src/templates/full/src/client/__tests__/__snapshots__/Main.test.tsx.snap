// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main message={string} /> renders correctly 1`] = `
<div>
  <h3>
    
  </h3>
</div>
`;
